### PR TITLE
GCW-3404 Performance improvement

### PR DESCRIPTION
**Pull request by bharat619**
### Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-3404

### What does this PR do?
- Performance optimisations
- Remove unused preload requests

### CODE REVIEW

File: app/components/goodcity/code-search-overlay.js [hunk 1]
```diff
@@ -22,6 +23,18 @@ export default Ember.Component.extend(SearchMixin, {
...
+    const hasData = this.get("store")
+      .peekAll("code")
+      .get("length");
+
+    if (this.get("open") && !hasData) {
+      await this.runTask(async () => {
+        await this.get("store").query("code", { stock: true });
+      }, ERROR_STRATEGIES.MODAL);
+    }
+  },
```
Comment (patrixr) [inline app/components/goodcity/code-search-overlay.js:36]: this kinda feels like the preload mechanism isn't doing its role properly. Also making requests in the background with no subsequent handling of the returned data feels a bit off

File: app/routes/items/detail.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ export default AuthorizeRoute.extend({
         }
       );
     }
+    this.store.findAll("restriction", { reload: true });
```
Comment (patrixr) [inline app/routes/items/detail.js:41]: you need to await the promises. As a general rule never have non-awaited promises unless it's intended

File: app/routes/orders/detail.js [hunk 3]
```diff
@@ -7,5 +7,9 @@ export default AuthorizeRoute.extend({
 
   model({ order_id }) {
     return this.loadIfAbsent("designation", order_id);
+  },
+
+  afterModel() {
+    this.store.query("cancellation_reason", { for: "order" });
```
Comment (patrixr) [inline app/routes/orders/detail.js:13]: same here, a non awaited promise is a dangerous promise :)

**Review by patrixr — APPROVED** [review]
Looks good, some non-awaited promises flagged though.

**Review by swatijadhav — APPROVED** [review]

**Review by shreyas098 — APPROVED** [review]

**Pull request merged**